code/modules/mob/living/carbon/human/species_types/dwarves.dm

```diff
@@ -0,0 +1,217 @@
+
+
+/datum/species/dwarf/on_species_loss(mob/living/carbon/H, datum/species/new_species)
+	. = ..()
+	H.transform = H.transform.Scale(1, 1.25) //And we undo it.
+	UnregisterSignal(H, COMSIG_MOB_SAY) //We register handle_speech is not being used.
+
+//Dwarf Name stuff
+/proc/dwarf_name() //hello caller: my name is urist mcuristurister
+	return "[pick(GLOB.dwarf_first)] [pick(GLOB.dwarf_last)]"
+
+/datum/species/dwarf/random_name(gender,unique,lastname)
+	return dwarf_name() //hello, ill return the value from dwarf_name proc to you when called.
+
+//Dwarf Speech handling - Basically a filter/forces them to say things. The IC helper
+/datum/species/dwarf/proc/handle_speech(datum/source, list/speech_args)
+	var/message = speech_args[SPEECH_MESSAGE]
+	if(message[1] != "*")
+		message = " [message]" //Credits to goonstation for the strings list.
+		var/list/dwarf_words = strings("dwarf_replacement.json", "dwarf") //thanks to regex too.
+
+		for(var/key in dwarf_words) //Theres like 1459 words or something man.
+			var/value = dwarf_words[key] //Thus they will always be in character.
+			if(islist(value)) //Whether they like it or not.
+				value = pick(value) //This could be drastically reduced if needed though.
+
+			message = replacetextEx(message, " [uppertext(key)]", " [uppertext(value)]")
+			message = replacetextEx(message, " [capitalize(key)]", " [capitalize(value)]")
+			message = replacetextEx(message, " [key]", " [value]") //Also its scottish.
+
+		if(prob(3))
+			message += pick(" By Armok!")
+	speech_args[SPEECH_MESSAGE] = trim(message)
+
+//This mostly exists because my testdwarf's liver died while trying to also not die due to no alcohol.
+/obj/item/organ/liver/dwarf
+	name = "dwarf liver"
+	icon_state = "liver"
+	desc = "A dwarven liver, theres something magical about seeing one of these up close."
+	alcohol_tolerance = 0 //dwarves really shouldn't be dying to alcohol.
+	toxTolerance = 5 //Shrugs off 5 units of toxins damage.
+	maxHealth = 150 //More health than the average liver, as you aren't going to be replacing this.
+	//If it does need replaced with a standard human liver, prepare for hell.
+
+//alcohol gland
+/obj/item/organ/dwarfgland
+	name = "dwarf alcohol gland"
+	icon_state = "plasma" //Yes this is a actual icon in icons/obj/surgery.dmi
+	desc = "A genetically engineered gland which is hopefully a step forward for humanity."
+	w_class = WEIGHT_CLASS_NORMAL
+	var/stored_alcohol = 250 //They start with 250 units, that ticks down and eventaully bad effects occur
+	var/max_alcohol = 500 //Max they can attain, easier than you think to OD on alcohol.
+	var/heal_rate = 0.5 //The rate they heal damages over 400 alcohol stored. Default is 0.5 so we times 3 since 3 seconds.
+	var/alcohol_rate = 10 //Its times 0.025 making it tick down by .25 per loop per 10. EX: 20 = .50
+	var/filth_counter = 0 //Holder for the filth check cycle, basically it compares against this.
```

Comment:
This is going to be used only on handle_filth_cycle() if the debug messages are removed.

Reply:
Its easier to adjust with admin var edit if its on the organ itself instead of in the cycle.

Reply:
Updated comment, Its there mostly so an admin can var edit it on the organ.

Reply:
Also note how the filth_counter doesn't reset to 0 each `handle_filth_cycle()` but only add up until it reaches amounts fatal for any dorf.
Var editing it won't have any impact ingame otherwise.

Reply:
Yeah, I can concede this one. Its more of a debug message at this point and there due to a old ticker iteration that attempted to use addtimer_varsets anyways